lib/networkinterfaces.js

```diff
@@ -24,7 +24,8 @@ module.exports = function () {
         var hasAddresses = false;
         ifaces[iface].forEach(function (address) {
             if (!address.internal) {
-                addresses[(address.family || "").toLowerCase()] = address.address;
+                var family = typeof address.family === 'number' ? "ipv" + address.family : address.family.toLowerCase() || "";
```

Comment:
I don't think this is equivalent to the previous code: It seems like it used to assume `address.family` could potentially be undefined: `(address.family || "").toLowerCase()`. With the new code it would fail at `address.family.toLowerCase()` if it were undefined.

It might not be an issue, but going from the code only this is what stands out to me?

Reply:
you are right
![image](https://user-images.githubusercontent.com/1029452/175333927-b73b3750-b4a4-4cb3-90db-8d848585e8a3.png)

Reply:
Good catch and good to see that other people look at what is going on in this repository too. Yes I did notice that in this pull request but didn't raise it, as it seemed to be faster to just do it myself. Could have pointed it out though and advertise that I was gonna fix it right away before releasing.

Then again that particular code path in question I think is happening only in some stone age node versions, or io.js, which you really shouldn't be using. Anyways, the guard is there and it doesn't hurt, so I maintain it (otherwise could just get rid of it).

Reply:
I just happened to pass by and noticed the discrepancy between the old logic and the new one.

Though looking more critically at this guard I don't know if it's really a good way of doing it?

If I understand correctly, the logic here dynamically adds fields to `addresses` (which is called `iface` where it is used [here](https://github.com/scravy/node-macaddress/blob/8deea57be8a4c404006d6a2885473a903fb92402/index.js#L53-L73)), and later a heuristic is testing said `iface` for fields like `ipv4` or `ipv6`. The current code would add an empty key to `iface` storing the address of "family-less" inputs...

In short it seems to me that if we still want some type of guard here then I'd write something like the following:

```js
        ifaces[iface].forEach(function (address) {
            if (!address.internal) {
                hasAddresses = true;
                if (typeof address.family === "number") { // node >=18
                    addresses["ipv" + address.family] = address.address;
                } else if (typeof address.family === "string") { // node <18
                    addresses[address.family.toLowerCase()] = address.address
                }
                if (address.mac && address.mac !== '00:00:00:00:00:00') {
                    addresses.mac = address.mac;
                }
            }
        });
```

This way if `address.family` isn't what we expect we won't get a weird unused empty string key on the result object. Unless this is by design?

Reply:
Looking at the site you're referencing, yepp, it's true – that empty string key is not being used. That's not by design, that's accidental. However, that function is exposed as part of the public interface. While I really do not expect it – who knows whether anyone uses it. So I'm inclined not to do anything about it.

However it seems to me what one could have been doing here is to actually look at the address and judge whether it's an ipv4 or an ipv6 address and set the family accordingly.

Ultimately though the guard can probably be removed. I do not think that modern node versions (any version remotely supported) will not return that field ever.

BTW I just checked the docs and node 18.4.0 is returning `.family` to string again...

Reply:
> BTW I just checked the docs and node 18.4.0 is returning .family to string again...  

🤦